tests/unit_tests/pyanaconda_tests/modules/localization/test_module_localization.py

```diff
@@ -648,69 +651,204 @@ def test_get_missing_keyboard_configuration_task(self, get_missing_mock):
 
         task = GetMissingKeyboardConfigurationTask(
             localed_wrapper=mocked_localed,
+            live_keyboard_wrapper=None,
             x_layouts="[cz (qwerty)]",
             vc_keymap="",
         )
         result = task.run()
         assert result == (x_layouts_result, vc_keymap_result)
 
+    @contextmanager
+    def _create_localed_mock(self,
+                             convert_layouts_output,
+                             convert_keymap_output,
+                             expected_convert_layouts_input=None,
+                             expected_convert_keymap_input=None):
+        localed = Mock()
+        localed.convert_layouts.return_value = convert_layouts_output
+        localed.convert_keymap.return_value = convert_keymap_output
+
+        yield localed
+
+        if expected_convert_layouts_input is None:
+            localed.convert_layouts.assert_not_called()
+        else:
+            localed.convert_layouts.assert_called_once_with(expected_convert_layouts_input)
+
+        if expected_convert_keymap_input is None:
+            localed.convert_keymap.assert_not_called()
+        else:
+            localed.convert_keymap.assert_called_once_with(expected_convert_keymap_input)
+
+    def _create_live_keyboard_mock(self, layouts, current_layout):
+        live_keyboard_mock = Mock()
+
+        live_keyboard_mock.read_keyboard_layouts.return_value = layouts
+        live_keyboard_mock.read_current_keyboard_layout.return_value = current_layout
+
+        return live_keyboard_mock
+
     def _get_missing_keyboard_configuration_test(self,
-                                                 x_layouts,
-                                                 converted_x_layouts,
-                                                 vc_keymap,
-                                                 converted_vc_keymap,
+                                                 input_x_layouts,
+                                                 input_vc_keymap,
                                                  result_x_layouts,
-                                                 result_vc_keymap):
-        localed = Mock()
-        localed.convert_keymap.return_value = converted_vc_keymap
-        localed.convert_layouts.return_value = converted_x_layouts
+                                                 result_vc_keymap,
+                                                 localed,
+                                                 live_keyboard):
 
         result = get_missing_keyboard_configuration(
```

Comment:
After that, you should be able to do here
```python
with patch("pyanaconda.modules.localization.utils.get_live_keyboard_instance") as get_mock:
    (...)
```
instead of passing the keyboard class as parameter.

Reply:
It's definitely less disruptive to the whole codebase.